### DISCUSSION TITLE
[Android] Events files (java/kotlin) are not generated

### DISCUSSION THREAD

**Discussion by troZee** [reactwg/react-native-new-architecture#73] [Q&A] [answered]
Hey all, I am trying to generate a component that exposes a callback as a prop. My codegen spec looks like
```
type Event = Readonly<{
  path?: string;
}>;

export interface NativeProps extends ViewProps {
  onCapture: DirectEventHandler<Event>;
}


export default codegenNativeComponent<NativeProps>('MyModule');
```

but when I ran `./gradlew generateCodegenArtifactsFromSchema` the output is
```
public interface MyModuleManagerInterface<T extends View> {
  // No props
}
```

EDIT: The above setup work fine on iOS, it does generate an EventEmitter.h with what seem a correct implementation
```
class MyModuleEventEmitter : public ViewEventEmitter {
 public:
  using ViewEventEmitter::ViewEventEmitter;

  struct OnCapture {
      std::string path;
    };

  void onCapture(OnCapture value) const;
};
```

Expected result: 
Keep platform consistency - we should not create events files manually like this: https://github.com/facebook/react-native/search?q=ReactSlidingCompleteEvent

**Comment by cortinico** [comment, marked as answer]
Yes you're right that we're not generating the corresponding Event class for Android in Java. That's a feature we can add to the codegen and we're actually open to receive contributions on it if someone is willing to take a stance at it 👍
